src/Service/DescartarCarrinhoService.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Service;
+
+use App\Controller\Exception\CarrinhoJaDescartadoException;
+use App\Entity\Carrinho;
+use App\Entity\StatusEnum;
+use App\Repository\CarrinhoRepository;
+
+class DescartarCarrinhoService
+{
+  public function __construct(
+    private CarrinhoRepository $carrinhoRepository,
+  ) {}
+
+  public function execute(int|Carrinho $carrinho): void
+  {
+    $carrinho = $this->carrinhoRepository->find($carrinho);
+
+    if ($carrinho->getStatus() === StatusEnum::descartado) {
```

Comment:
cria um método assim dentro da entidade carrinho e lá faz o return `$this->getStatus() === StatusEnum::descartado`
```suggestion
    if ($carrinho->isDescartado()) {
```

Reply:
faz o mesmo para isAberto